.github/actions/slack/action.yml

```diff
@@ -0,0 +1,31 @@
+name: "Slack Notification for Test Failures"
+
+inputs:
+  python_result:
+    description: 'Python Test Result'
+  javascript_result:
+    description: 'Javascript Test Result'
+  py_prod_deps_result:
+    description: 'Python Prod Deps Test Result'
+  cypress_result:
+    description: 'Cypress Test Result'
+
+runs:
+  using: composite
+  steps:
+    - if: ${{ inputs.python_result == 'failure' }}
+      run: |
+        curl -X POST -H 'Content-type: application/json' --data '{"text":":blobonfire: Nightly build failed on python tests"}' "$SLACK_WEBHOOK"
```

Comment:
Whoops, not sure how I didn't realize this earlier, but won't the slack webhook URL be printed when the `curl` command being run is echoed here?

We could turn off commands being echoed before being run, but it might be safer to write a thin wrapper Python script that posts to the webhook and pulls the webhook URL out of an environment variable to make it harder to mess up and accidentally end up logging a secret.

Reply:
Good news, GHA automatically redacts secrets printed to the log, so the webhook URL isn't shown, but agree a refactor might make sense for the slack messages.
Example w/ slack post as step:
<img width="899" alt="Screen Shot 2022-09-28 at 5 05 55 PM" src="https://user-images.githubusercontent.com/63436329/192910659-04394025-9131-4c31-bce5-23344e415234.png">
Example w/ slack post as action:
<img width="845" alt="Screen Shot 2022-09-28 at 5 02 22 PM" src="https://user-images.githubusercontent.com/63436329/192910815-691594e2-f84c-4e14-99a7-d35eb9f74b3e.png">